docs/sql-migration-guide-upgrade.md

```diff
@@ -7,6 +7,12 @@ displayTitle: Spark SQL Upgrading Guide
 * Table of contents
 {:toc}
 
+## Upgrading from Spark SQL 2.4 to 2.4.5
+
+ - Starting from 2.4.5, SQL configurations are effective also when a Dataset is converted to an RDD and its
+   plan is executed due to action on the derived RDD. The previous buggy behavior can be restored setting
+   `spark.sql.legacy.rdd.applyConf` to `false`.
```

Comment:
Hi, @hvanhovell , @cloud-fan and @gatorsmile . As @mgaido91 asked here, this PR will add this flag only at `branch-2.4`. In this case, is it okay this config will be **added and deprecated** at `2.4.5` and will be removed at `3.0.0`?

For me, we don't need to add this configuration to `master`. Did I understand correctly?